### ISSUE TITLE
both numerical and categorical inputs

### ISSUE THREAD

**Issue by Lauge1006** [rstudio/keras#711]
I want to create a NN with (lets say) 2 numerical inputs (scaled to 0-1) and 2 categorcial inputs (integers). I know that a categorical value can be embedded to a 0-1 vector by the layer `layer_embedding`, i can also do that on the other categorical value and then concatenate the NN's with `layer_concatenate`

But when i want to concatenate with the numerical values the concatenation has a problem. The `layer_concatenate` requires inputs to have same shape except for concat axis, which is impossible since the numerical values are a 2->2 tensor (identity function), but the embeddings are a 1->20 and 1->20 (if there are 20 categories)

How to you make is into **one** network, that after this does standard `layer_dense` stuff? (eg one that takes in these 4 inputs). Maybe the `layer_concatenate` is not the right solution, but surely there must be a way to do this :)

**Comment by henry090** [comment]
Hi, @ skeydan wrote a great post on tensorflow blog (https://blogs.rstudio.com/tensorflow/posts/2018-11-26-embeddings-fun-and-profit/) where you can find a solution for your problem. Please look at the end of the post (link) in order to find the code snippets

```

prod_input <- layer_input(shape = 1)
cont_input <- layer_input(shape = 2)

prod_embed <- prod_input %>% 
  layer_embedding(input_dim = sales_embed$Prod %>% max() + 1,
                  output_dim = 256
  ) %>%
  layer_flatten()
cont_dense <- cont_input %>% layer_dense(units = 256, activation = "selu")

output <- layer_concatenate(
  list(prod_embed, cont_dense)) %>%
  layer_dropout(dropout_rate) %>% 
  layer_dense(units = 256, activation = "selu") %>%
  layer_dropout(dropout_rate) %>% 
  layer_dense(units = 256, activation = "selu") %>%
  layer_dropout(dropout_rate) %>% 
  layer_dense(units = 256, activation = "selu") %>%
  layer_dropout(dropout_rate) %>% 
  layer_dense(units = 1, activation = "sigmoid")

model <- keras_model(inputs = list(prod_input, cont_input), outputs = output)

model %>% compile(loss = "binary_crossentropy", optimizer = "adam", metrics = "accuracy")

model %>% fit(
  list(X_train[ , 1], X_train[ , 2:3]),
  y_train,
  validation_data = list(list(X_valid[ , 1], X_valid[ , 2:3]), y_valid),
  class_weights = list("0" = 0.1, "1" = 0.9),
  batch_size = 128,
  epochs = 200
)

```
